### ISSUE TITLE
Fail to install on OSX Sierra

### ISSUE THREAD

**Issue by adamist521** [BurntSushi/erd#24]
Failed to install erd with cabal.

```
$ cabal install erd
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: erd-0.1.3.0 (user goal)
next goal: base (dependency of erd-0.1.3.0)
rejecting: base-4.9.1.0/installed-4.9... (conflict: erd => base==4.7.*)
rejecting: base-4.9.1.0, base-4.9.0.0, base-4.8.2.0, base-4.8.1.0,
base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0, base-4.6.0.1,
base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0, base-4.4.0.0,
base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1, base-4.2.0.0,
base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1 (constraint from
non-upgradeable package requires installed instance)
Dependency tree exhaustively searched.
$
```
FYI, 
```
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.12.4
BuildVersion:	16E195

**Comment by adamist521** [comment]
Was able to install by code below.

```
$ git clone git://github.com/BurntSushi/erd
$ cd erd
$ stack init
$ stack build --system-ghc
```

**Comment by qizwiz** [comment]
Thanks for this!
While I think your method is preferable, I was also able to get it to install by changing the base from 4.7 to 4.9 in the cabal file.

**Comment by chreble** [comment]
```cabal install --allow-newer erd``` also works

**Comment by carld** [comment]
Haskell newbie here, started out with `brew cask install haskell-platform`

```
% cabal install --allow-newer erd
Resolving dependencies...
Configuring erd-0.1.3.0...
Building erd-0.1.3.0...
Failed to install erd-0.1.3.0
Build log ( /Users/carldouglas/.cabal/logs/ghc-8.2.1/erd-0.1.3.0-8kV5AS2coBX939FGaGyt2n.log ):
cabal: Entering directory '/var/folders/jy/_s_btrwn7c99qnyw7d0t7dtw0000gn/T/cabal-tmp-61706/erd-0.1.3.0'
Configuring erd-0.1.3.0...
Preprocessing executable 'erd' for erd-0.1.3.0..
Building executable 'erd' for erd-0.1.3.0..
<command line>: cannot satisfy -package-id parsec-3.1.11-DPgnR92AWEaFOaixmwipet: 
    parsec-3.1.11-DPgnR92AWEaFOaixmwipet is unusable due to shadowed dependencies:
      mtl-2.2.1-19EL8AGBsN3DnnOhrC9xY3 text-1.2.2.2-EGUst8sqNAZCw1xLPcmcMH
    (use -v for more information)
cabal: Leaving directory '/var/folders/jy/_s_btrwn7c99qnyw7d0t7dtw0000gn/T/cabal-tmp-61706/erd-0.1.3.0'
cabal: Error: some packages failed to install:
erd-0.1.3.0-8kV5AS2coBX939FGaGyt2n failed during the building phase. The
exception was:
ExitFailure 1
```
Any suggestions appreciated thanks!

**Comment by carld** [comment]
For posterity... I installed erd after running `--force-reinstalls` of parsec etc

**Comment by combinatorist** [comment]
I spent 3 hours trying to install this package. As a first time haskell user, I tried everything with no clue what I was doing until I saw this comment: https://github.com/BurntSushi/erd/issues/24#issuecomment-314775206 by @chreble

Thanks @chreble!

I just made a PR (#27) to "fix" the documentation. I have no idea how this works, but would love to make it easier for first time users.

**Comment by kyr0** [comment]
Yes, please add it to the docs; it's such an easy change... I was looking into it for a few minutes too.

**Comment by benjamin-demarteau** [comment]
Still having problems, there seems to be an error with wl-pprinter-text:
```
➜  ~ cabal install --allow-newer --force-reinstalls erd
Resolving dependencies...
Configuring wl-pprint-text-1.1.1.0...
Building wl-pprint-text-1.1.1.0...
Failed to install wl-pprint-text-1.1.1.0
Build log ( /Users/benjamin/.cabal/logs/ghc-8.4.1/wl-pprint-text-1.1.1.0-7Rq3bROoVOxBg93W8s5ShZ.log ):
cabal: Entering directory '/var/folders/5t/6hpmgtb91t15s5b9k_zj35ch0000gn/T/cabal-tmp-57070/wl-pprint-text-1.1.1.0'
Configuring wl-pprint-text-1.1.1.0...
Preprocessing library for wl-pprint-text-1.1.1.0..
Building library for wl-pprint-text-1.1.1.0..
[1 of 2] Compiling Text.PrettyPrint.Leijen.Text ( Text/PrettyPrint/Leijen/Text.hs, dist/build/Text/PrettyPrint/Leijen/Text.o )

Text/PrettyPrint/Leijen/Text.hs:122:1: error:
    Could not find module ‘Prelude.Compat’
    There are files missing in the ‘base-compat-0.9.3’ package,
    try running 'ghc-pkg check'.
    Use -v to see a list of the files searched for.
    |
122 | import Prelude.Compat hiding ((<$>))
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Leaving directory '/var/folders/5t/6hpmgtb91t15s5b9k_zj35ch0000gn/T/cabal-tmp-57070/wl-pprint-text-1.1.1.0'
cabal: Error: some packages failed to install:
erd-0.1.3.0-8kV5AS2coBX939FGaGyt2n depends on erd-0.1.3.0 which failed to
install.
graphviz-2999.19.0.0-JwaTXCIUDJJPYkg3249Wq depends on graphviz-2999.19.0.0
which failed to install.
wl-pprint-text-1.1.1.0-7Rq3bROoVOxBg93W8s5ShZ failed during the building
phase. The exception was:
ExitFailure 1
```

**Comment by nwokeo** [comment]
I had the same problem, seems to be due to [this issue](https://github.com/ivan-m/wl-pprint-text/issues/18). 

I was able to resolve by installing haskell-platform via cask: `brew cask install haskell-platform` then installing erd: `cabal install --allow-newer erd`

**Comment by benjamin-demarteau** [comment]
> I was able to resolve by installing haskell-platform via cask: brew cask install haskell-platform then installing erd: cabal install --allow-newer erd

That fixed it for me, thanks.

**Comment by rclayton-the-terrible** [comment]
I don't know how this is working for people.  I've tried on OSX and Docker and keep getting the following output:

```
Data/GraphViz/Printing.hs:232:16: error:
    • Variable not in scope: (<>) :: DotCode -> DotCode -> t2
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
232 | wrap b a d = b <> d <> a
    |                ^^

Data/GraphViz/Printing.hs:232:21: error:
    • Variable not in scope: (<>) :: t2 -> DotCode -> DotCode
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
232 | wrap b a d = b <> d <> a
    |                     ^^

Data/GraphViz/Printing.hs:235:26: error:
    • Variable not in scope: (<>) :: DotCode -> m2 Doc -> t1
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
235 | commaDel a b = unqtDot a <> comma <> unqtDot b
    |                          ^^

Data/GraphViz/Printing.hs:235:35: error:
    • Variable not in scope: (<>) :: t1 -> DotCode -> DotCode
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
235 | commaDel a b = unqtDot a <> comma <> unqtDot b
    |                                   ^^

Data/GraphViz/Printing.hs:238:25: error:
    • Variable not in scope: (<>) :: m0 Doc -> m1 Doc -> t0
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
238 | printField f v = text f <> equals <> toDot v
    |                         ^^

Data/GraphViz/Printing.hs:238:35: error:
    • Variable not in scope: (<>) :: t0 -> DotCode -> DotCode
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
238 | printField f v = text f <> equals <> toDot v
    |                                   ^^

Data/GraphViz/Printing.hs:295:37: error:
    • Variable not in scope: (<>) :: DotCode -> DotCode -> DotCode
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude),
        ‘PP.<$>’ (imported from Text.PrettyPrint.Leijen.Text.Monadic),
        ‘*>’ (imported from Prelude)
    |
295 |   unqtDot (BScheme n l) = unqtDot n <> unqtDot l
    |                                     ^^
cabal: Leaving directory '/tmp/cabal-tmp-126/graphviz-2999.19.0.0'
cabal: Error: some packages failed to install:
erd-0.1.3.0-8kV5AS2coBX939FGaGyt2n depends on erd-0.1.3.0 which failed to
install.
graphviz-2999.19.0.0-Hxa68Q9ziKb1pHI4Xvs5hL failed during the building phase.
The exception was:
ExitFailure 1
```

**Comment by qizwiz** [comment]
@rclayton-the-terrible I just tried it myself and got your same errors. I noticed all the errors were the same and from the same package.

From that, I tried installing graphviz by itself. 

`$ cabal install graphviz`

That worked. And then I was able to do:

`$ cabal install --allow-newer erd`

So give that a shot.

**Comment by rclayton-the-terrible** [comment]
@qizwiz hey that worked!  Thank you so much.

I don't know a thing about Haskell or it's toolchain, but I noticed installing via `cabal` doesn't create an executable on the `$PATH`, so I've been running everything with `cabal run erd -- ...` . Is this the right approach?  Or am I missing something?

**Comment by qizwiz** [comment]
@rclayton-the-terrible 
I have a mac. It looks like you have linux.

The relevant lines in my ~/.cabal/config file looks like this

```
-- === Built executables will be installed in:
--     ~/Library/Haskell/bin
--
-- You may wish to place this on your PATH by adding the following
-- line to your ~/.bash_profile:
--     export PATH="$HOME/Library/Haskell/bin:$PATH"

```
If this file exists, perhaps you'll find the same hint there for you (it'll almost certainly be different).
The documentation can be found here: [2.1. Configuration](https://www.haskell.org/cabal/users-guide/installing-packages.html)

**Comment by rclayton-the-terrible** [comment]
@qizwiz I tried both macOS and Linux (as a Docker container `FROM haskell`.  Same outcome for both.  The fix was to install Graphviz via `cabal` and not some other mechanism (like `brew`).

**Comment by qizwiz** [comment]
@rclayton-the-terrible I was addressing the second part of your question. Seemed like you were having problems with the binpath for packages you installed with cabal
